gateway/src/test/java/io/camunda/zeebe/gateway/impl/probes/health/ClusterHealthIndicatorTest.java

```diff
@@ -0,0 +1,104 @@
+import org.junit.Test;
+import org.springframework.boot.actuate.health.Status;
+
+public class ClusterHealthIndicatorTest {
+
+  @Test
+  public void shouldRejectNullInConstructor() {
+    // when + then
+    assertThatThrownBy(() -> new ClusterHealthIndicator(null))
+        .isInstanceOf(NullPointerException.class);
+  }
+
+  @Test
+  public void shouldReportDownIfSupplierReturnsEmpty() {
+    // given
+    final Supplier<Optional<BrokerClusterState>> stateSupplier = () -> Optional.empty();
+    final var sutHealthIndicator = new ClusterHealthIndicator(stateSupplier);
+
+    // when
+    final var actualHealth = sutHealthIndicator.health();
+
+    // then
+    assertThat(actualHealth).isNotNull();
+    assertThat(actualHealth.getStatus()).isEqualTo(Status.DOWN);
+  }
+
+  @Test
+  public void shouldReportDownIfListOfBrokersAnIsNotEmptyAndPartitionsIsEmpty() {
+    // given
+    final BrokerClusterState mockClusterState = mock(BrokerClusterState.class);
+    when(mockClusterState.getBrokers()).thenReturn(List.of(1));
+    when(mockClusterState.getPartitions()).thenReturn(List.of());
+
+    final Supplier<Optional<BrokerClusterState>> stateSupplier =
+        () -> Optional.of(mockClusterState);
+    final var sutHealthIndicator = new ClusterHealthIndicator(stateSupplier);
+
+    // when
+    final var actualHealth = sutHealthIndicator.health();
+
+    // then
+    assertThat(actualHealth).isNotNull();
+    assertThat(actualHealth.getStatus()).isEqualTo(Status.DOWN);
+  }
+
+  @Test
+  public void shouldReportUpIfHasHealthyPartition() {
+    // given
+    final BrokerClusterState mockClusterState = mock(BrokerClusterState.class);
+    when(mockClusterState.getBrokers()).thenReturn(List.of(1));
+    when(mockClusterState.getPartitions()).thenReturn(List.of(1));
+    when(mockClusterState.getLeaderForPartition(1)).thenReturn(1);
+    when(mockClusterState.getPartitionHealth(1, 1)).thenReturn(PartitionHealthStatus.HEALTHY);
+
+    final Supplier<Optional<BrokerClusterState>> stateSupplier =
+        () -> Optional.of(mockClusterState);
+    final var sutHealthIndicator = new ClusterHealthIndicator(stateSupplier);
+
+    // when
+    final var actualHealth = sutHealthIndicator.health();
+
+    // then
+    assertThat(actualHealth).isNotNull();
+    assertThat(actualHealth.getStatus()).isEqualTo(Status.UP);
+  }
+
+  @Test
+  public void shouldReportDownIfListOfBrokersIsEmpty() {
+    // given
+    final BrokerClusterState mockClusterState = mock(BrokerClusterState.class);
+    when(mockClusterState.getBrokers()).thenReturn(emptyList());
+
+    final Supplier<Optional<BrokerClusterState>> stateSupplier =
+        () -> Optional.of(mockClusterState);
+    final var sutHealthIndicator = new ClusterHealthIndicator(stateSupplier);
+
+    // when
+    final var actualHealth = sutHealthIndicator.health();
+
+    // then
+    assertThat(actualHealth).isNotNull();
+    assertThat(actualHealth.getStatus()).isEqualTo(Status.DOWN);
+  }
+}
```

Comment:
:x: Also add following test cases
1. Should return degraded if there are missing partitions, all other partitions are healthy
2. Should return down if there are missing partitions, and all other partitions are unhealthy
3. Should return degraded if some partitions doesn't have a leader.